samples/modules/lvgl/acceleration_chart/src/main.c

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2023 Benjamin Cabé <benjamin@zephyrproject.org>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/display.h>
+#include <zephyr/drivers/sensor.h>
+
+#include <lvgl.h>
+#include <stdio.h>
+#include <string.h>
+#include <zephyr/kernel.h>
+
+#define LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(app);
+
+static lv_obj_t *chart1;
+static lv_chart_series_t *ser_x;
+static lv_chart_series_t *ser_y;
+static lv_chart_series_t *ser_z;
+lv_timer_t *fetch_and_display_timer;
+
+#define ACCEL0_NODE DT_ALIAS(accel0)
+const struct device *display_dev;
+const struct device *sensor;
+
+
+/* Timer handler: fetches sensor data and append it to the chart */
+static void fetch_and_display(lv_timer_t *timer)
+{
+	struct sensor_value accel[3];
+	int rc = sensor_sample_fetch(sensor);
+
+	if (rc == 0) {
+		rc = sensor_channel_get(sensor, SENSOR_CHAN_ACCEL_XYZ, accel);
+	}
+	if (rc < 0) {
+		LOG_ERR("ERROR: Update failed: %d\n", rc);
+	} else {
+		lv_chart_set_next_value(chart1, ser_x, sensor_value_to_double(&accel[0]));
+		lv_chart_set_next_value(chart1, ser_y, sensor_value_to_double(&accel[1]));
+		lv_chart_set_next_value(chart1, ser_z, sensor_value_to_double(&accel[2]));
+	}
+}
+
+void accelerometer_chart(lv_obj_t *parent)
```

Comment:
Nit: personal prefrence but I think there is a verb missing in the function name, maybe `init` or  `create`.

Reply:
oh wow yeah, totally agreed - so much for hastily cleaning up a very old PoC into a sample :)

Reply:
i went for `create_...` as this seems to be what's used for the demos.